### PR TITLE
test: Update OIDC integration test

### PR DESCRIPTION
**Pull request by labrenbe**
# Description

Part of https://github.com/stackabletech/druid-operator/issues/473

Update the OIDC integration test to do a full OIDC login flow and check that data ingestion is not broken.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [x] (Integration-)Test cases added and/or extended
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

**Review by adwk67 — CHANGES_REQUESTED** [review]
OIDC test runs on Kind and Openshift with the suggested change.

File: tests/templates/kuttl/oidc/51-install-test-container.yaml.j2 [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+          env:
+            - name: REQUESTS_CA_BUNDLE
+              value: /stackable/tls/ca.crt
+            - name: DRUID_OIDC
+              value: "true"
+            - name: DRUID_USER
+              value: druid_system
+            - name: DRUID_PASSWORD
+              value: druid-system-password
+{% if test_scenario['values']['oidc-use-tls'] == 'true' %}
```
[the remark this answers is not in the document]
Comment (labrenbe) [reply tests/templates/kuttl/oidc/51-install-test-container.yaml.j2]: I've updated the PR with your proposed change

[the remark this answers is not in the document]
Comment (adwk67) [reply tests/templates/kuttl/oidc/51-install-test-container.yaml.j2]: Thanks!

**Review by adwk67 — APPROVED** [review]

**Comment by sbernauer** [comment]
@labrenbe the nightly tests are failing wuth `KeyError: 'DRUID_OIDC'`, do you mind having a look?
https://ci.stackable.tech/view/01%20Operator%20Tests%20(nightly)/job/druid-operator-it-nightly/520/

**Pull request merged**